Projects/My_website/myWebsite/blog/views.py

```diff
@@ -105,20 +104,33 @@ def partial_update(self, request, pk):
 """ class to like a blog"""
 
 
-class LikeBlog(viewsets.ModelViewSet):
-    queryset = Like.objects.all()
+class LikeTheBlogTest(APIView):
     authentication_classes = [TokenAuthentication]
     permission_classes = [IsAuthenticated]
-    serializer_class = LikeBlogSerializer
+
+    def post(self, request):
+        try:
+            blog = Post.objects.get(id=request.data["blog-id"])
+            obj = blog.like.filter(id=request.user.id)
+            if obj.exists():
+                blog.like.remove(request.user.id)
+                return Response({"you disliked the blog"})
+            blog.like.add(request.user.id)
+            blog.save()
+            return Response({"you liked this blog"})
+        except:
+            return Response({"message": "Provided blog-id is not valid"})
```

Comment:
@codal-uchoudhary 
Don't provide a static response. Raise the exception.

Reply:
Implemented